snapcraft/tests/test_commands_list_plugins.py

```diff
@@ -23,25 +24,45 @@
 
 class ListPluginsCommandTestCase(tests.TestCase):
 
+    # plugin list when wrapper at MAX_CHARACTERS_WRAP
+    default_plugin_output = (
+        "ant        catkin  copy  jdk     kernel  maven  nodejs   python3  tar-content\n"  # noqa
+        "autotools  cmake   go    kbuild  make    nil    python2  scons  \n")  # noqa
+
+    @mock.patch('sys.stdout', new_callable=io.StringIO)
+    @mock.patch('subprocess.check_output')
+    def test_list_plugins_large_terminal(self, mock_subprocess, mock_stdout):
+        mock_subprocess.return_value = "999"
+        main(['list-plugins'])
+        self.assertEqual(mock_stdout.getvalue(), self.default_plugin_output)
+
+    @mock.patch('sys.stdout', new_callable=io.StringIO)
+    @mock.patch('subprocess.check_output')
+    def test_list_plugins_small_terminal(self, mock_subprocess, mock_stdout):
+        mock_subprocess.return_value = "60"
+        expected_output = (
+            "ant        cmake  jdk     make   nodejs   scons      \n"
+            "autotools  copy   kbuild  maven  python2  tar-content\n"
+            "catkin     go     kernel  nil    python3\n")
+        main(['list-plugins'])
+        self.assertEqual(mock_stdout.getvalue(), expected_output)
+
+    @mock.patch('sys.stdout', new_callable=io.StringIO)
+    @mock.patch('subprocess.check_output')
+    def test_list_plugins_error_invalid_terminal_size(self, mock_subprocess,
+                                                      mock_stdout):
+        def raise_error(x):
+            raise OSError()
+        mock_subprocess.side_effect = raise_error
+        main(['list-plugins'])
```

Comment:
oh, there's a similar test in `integration_tests` reason for adt to fail most likely

`runtests.sh integration` ftw ;-)

Or `python3 -m unittest integration_tests ...`